tests/integration/utils.py

```diff
@@ -255,6 +264,12 @@ def fetch_deps_and_check_output(
         expected_files_checksums = _load_json(
             test_data_dir.joinpath(test_data_dir, test_case, "fetch_deps_sha256sums.json")
         )
+
+        _update_test_data_if_needed(
+            test_data_dir.joinpath(test_data_dir, test_case, "fetch_deps_sha256sums.json"),
+            files_checksums,
+        )
+
         log.info("Compare checksums of fetched deps files")
         assert files_checksums == expected_files_checksums
```

Comment:
If you're in data-generating mode and the previous expected data is wrong, the test fails. IMO that's confusing. In generating mode, I would not expect the test to fail when checking the data that it just generated.